airbyte-webapp/src/pages/ConnectorBuilderPage/ConnectorBuilderPage.tsx

```diff
@@ -1,61 +1,93 @@
-import { builderFormValidationSchema } from "components/connectorBuilder/types";
+import { builderFormValidationSchema, BuilderFormValues } from "components/connectorBuilder/types";
 import { YamlEditor } from "components/connectorBuilder/YamlEditor";
 import { ResizablePanels } from "components/ui/ResizablePanels";
 
 import {
+  ConnectorBuilderAPIProvider,
   ConnectorBuilderStateProvider,
   useConnectorBuilderState,
 } from "services/connectorBuilder/ConnectorBuilderStateService";
 
 import styles from "./ConnectorBuilderPage.module.scss";
 
-const ConnectorBuilderPageInner: React.FC = () => {
-  const { formatMessage } = useIntl();
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = function () {};
+
+const ConnectorBuilderPageInner: React.FC = React.memo(() => {
   const { builderFormValues, editorView, setEditorView } = useConnectorBuilderState();
 
-  return (
-    <Formik initialValues={builderFormValues} onSubmit={() => undefined} validationSchema={builderFormValidationSchema}>
-      {({ values }) => (
-        <ResizablePanels
-          className={classnames({ [styles.gradientBg]: editorView === "yaml", [styles.solidBg]: editorView === "ui" })}
-          firstPanel={{
-            children: (
-              <>
-                {editorView === "yaml" ? (
-                  <YamlEditor toggleYamlEditor={() => setEditorView("ui")} />
-                ) : (
-                  <Builder values={values} toggleYamlEditor={() => setEditorView("yaml")} />
-                )}
-              </>
-            ),
-            className: styles.leftPanel,
-            minWidth: 100,
-          }}
-          secondPanel={{
-            children: <StreamTestingPanel />,
-            className: styles.rightPanel,
-            flex: 0.33,
-            minWidth: 60,
-            overlay: {
-              displayThreshold: 325,
-              header: formatMessage({ id: "connectorBuilder.testConnector" }),
-              rotation: "counter-clockwise",
-            },
-          }}
-        />
-      )}
-    </Formik>
+  const switchToUI = useCallback(() => setEditorView("ui"), [setEditorView]);
+  const switchToYaml = useCallback(() => setEditorView("yaml"), [setEditorView]);
+
+  const initialFormValues = useRef(builderFormValues);
+  return useMemo(
+    () => (
+      <Formik initialValues={initialFormValues.current} onSubmit={noop} validationSchema={builderFormValidationSchema}>
+        {({ values }) => (
+          <Panels editorView={editorView} switchToUI={switchToUI} values={values} switchToYaml={switchToYaml} />
+        )}
+      </Formik>
+    ),
+    [editorView, switchToUI, switchToYaml]
   );
-};
+});
 
 export const ConnectorBuilderPage: React.FC = () => (
   <ConnectorBuilderStateProvider>
-    <ConnectorBuilderPageInner />
+    <ConnectorBuilderAPIProvider>
+      <ConnectorBuilderPageInner />
+    </ConnectorBuilderAPIProvider>
   </ConnectorBuilderStateProvider>
 );
+
+const Panels = React.memo(
```

Comment:
The motivation behind moving this into it's own component is so that we can wrap it in a React.memo(), correct?

Reply:
Exactly, as validation is async in formik this re-renders multiple times which I try to contain as much as possible